src/common/android/androidPlatform.ts

```diff
@@ -0,0 +1,177 @@
+                () =>
+                    Q(AndroidPlatform.RUN_ANDROID_SUCCESS_PATTERNS),
+                () =>
+                    Q(AndroidPlatform.RUN_ANDROID_FAILURE_PATTERNS)).process(runAndroidSpawn);
+
+            return output
+                .finally(() => {
+                    return this.initializeTargetDevicesAndPackageName();
+                }).then(() => [this.debugTarget], reason => {
+                    if (reason.message === AndroidPlatform.MULTIPLE_DEVICES_ERROR && this.devices.length > 1 && this.debugTarget) {
+                        /* If it failed due to multiple devices, we'll apply this workaround to make it work anyways */
+                        this.needsToLaunchApps = true;
+                        return shouldLaunchInAllDevices
+                            ? this.adb.getOnlineDevices()
+                            : Q([this.debugTarget]);
+                    } else {
+                        return Q.reject<IDevice[]>(reason);
+                    }
+                }).then(devices => {
+                    return new PromiseUtil().forEach(devices, device => {
+                        return this.launchAppWithADBReverseAndLogCat(device);
+                    });
+                });
+        });
+    }
+
+    public enableJSDebuggingMode(): Q.Promise<void> {
+        return this.adb.reloadAppInDebugMode(this.runOptions.projectRoot, this.packageName, this.debugTarget.id);
+    }
+
+    private initializeTargetDevicesAndPackageName(): Q.Promise<void> {
+        return this.adb.getConnectedDevices().then(devices => {
+            this.devices = devices;
+            this.debugTarget = this.getTargetEmulator(devices);
+            return this.getPackageName().then(packageName => {
+                this.packageName = packageName;
+            });
+        });
+    }
+
+    private launchAppWithADBReverseAndLogCat(device: IDevice): Q.Promise<void> {
+        return Q({})
+            .then(() => {
+                return this.configureADBReverseWhenApplicable(device);
+            }).then(() => {
+                return this.needsToLaunchApps
+                    ? this.adb.launchApp(this.runOptions.projectRoot, this.packageName, device.id)
+                    : Q<void>(void 0);
+            }).then(() => {
+                return this.startMonitoringLogCat(device, this.runOptions.logCatArguments).catch(error => // The LogCatMonitor failing won't stop the debugging experience
+                    Log.logWarning("Couldn't start LogCat monitor", error));
+            });
+    }
+
+    private configureADBReverseWhenApplicable(device: IDevice): Q.Promise<void> {
+        if (device.type !== DeviceType.AndroidSdkEmulator) {
+            return Q({}) // For other emulators and devices we try to enable adb reverse
+                .then(() => this.adb.apiVersion(device.id))
+                .then(apiVersion => {
+                    if (apiVersion >= AndroidAPILevel.LOLLIPOP) { // If we support adb reverse
```

Comment:
I can change it to a global variable if you think that's better. But I want to use a string, instead of a number so it's easy to understand what's happening